src/camera.js

```diff
@@ -590,6 +648,46 @@ vgl.camera = function () {
 
   ////////////////////////////////////////////////////////////////////////////
   /**
+   * Check if the texture should be aligned to the screen.  Alignment only
+   * occurs if the parallel extents contain width, height, and a
+   * close-to-integer zoom level, and if the units-per-pixel value has been
+   * computed.  The camera must either be in parallel projection mode OR must
+   * have a perspective camera which is oriented along the z-axis without any
+   * rotation.
+   *
+   * @returns: either null if no alignment should occur, or an alignment object
+   *           with the rounding value and offsets.
+   */
+  ////////////////////////////////////////////////////////////////////////////
+  this.viewAlignment = function () {
+    if (!m_enableParallelProjection) {
+      /* If we aren't in parallel projection mode, ensure that the projection
+       * matrix meets strict specifications */
+      var proj = this.projectionMatrix();
+      if (proj[1] || proj[2] || proj[3] || proj[4] || proj[6] || proj[7] ||
+          proj[8] || proj[9] || proj[12] || proj[13] || proj[15]) {
+        return null;
+      }
+    }
+    if (!m_parallelExtents[0] || !m_parallelExtents[1] || !m_unitsPerPixel) {
+      return null;
+    }
+    if (parseFloat(m_parallelExtents[2].toFixed(6)) !==
+        parseFloat(m_parallelExtents[2].toFixed(0))) {
+      return null;
+    }
+    var align = {round: m_unitsPerPixel, dx: 0, dy: 0};
+    if (m_parallelExtents[0] % 2) {
```

Comment:
Adding some explanations might be helpful here specially when we look back at this code in the future.